### PR TITLE
Upgrading to Unity 2018.2.0b10

### PR DESCRIPTION
**Pull request by jessicafalk**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Upgrading to 2018.2.0b10

Important note: `ReadToEnd` throws an IOException in this version. It seems to be related to having an empty line at the end or some invalid character. When reading line by line, it seems to work fine.

#### Tests
Ran local and cloud deployments.

#### Documentation
Updated documentation to this beta version.

#### Primary reviewers
@jamiebrynes7 @zeroZshadow

### CODE REVIEW

File: workers/unity/Assets/Gdk/Legacy/Editor/BuildSystem/Util/SpatialCommandRunner.cs [hunk 1]
```diff
@@ -27,5 +28,24 @@ internal static void RunSpatialCommand(string args, string description)
...
+            while (line != null)
+            {
+                try
+                {
+                    line = reader.ReadLine();
+                    output += line;
+                }
+                catch
+                {
+                    line = null;
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Gdk/Legacy/Editor/BuildSystem/Util/SpatialCommandRunner.cs]: Just `break` out of the while loop?

File: workers/unity/Packages/manifest.json [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
   "dependencies": {
     "com.unity.entities": "0.0.12-preview.3",
-    "com.unity.ads": "exclude",
-    "com.unity.purchasing": "exclude",
-    "com.unity.analytics": "exclude",
-    "com.unity.standardevents": "exclude",
-    "com.unity.postprocessing": "2.0.5-preview"
+    "com.unity.package-manager-ui": "1.9.9",
+    "com.unity.postprocessing": "2.0.5-preview",
+    "com.unity.textmeshpro": "1.2.1",
```
[the remark this answers is not in the document]
Comment (jessicafalk) [reply workers/unity/Packages/manifest.json]: Actually: most of them are built in packages.  Can only disable them, which would already be a good thing.

File: workers/unity/Assets/Gdk/Legacy/Editor/BuildSystem/Util/SpatialCommandRunner.cs (hunk 1 above)
Comment (jessicafalk) [reply workers/unity/Assets/Gdk/Legacy/Editor/BuildSystem/Util/SpatialCommandRunner.cs]: changed the whole code after speaking to @jared-improbable

File: workers/unity/Packages/manifest.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (jessicafalk) [reply workers/unity/Packages/manifest.json]: done! left only those in that are absolutely needed

**Comment by jessicafalk** [comment]
as Unity updated their beta to 10 yesterday, I'll leave this PR open until I upgraded everything to the latest beta and then update this PR

File: docs/setup-and-installing.md [hunk 3]
```diff
@@ -23,7 +23,7 @@
 ### Setting up your machine
 
 1. Install prerequisites:
-	- [Unity 2018.1.1](https://unity3d.com/get-unity/download/archive)
+	- [Unity 2018.2.0b10](https://unity3d.com/unity/beta-download)
```
Comment (oblm) [inline docs/setup-and-installing.md]: Good job finding [/beta-download](https://unity3d.com/unity/beta-download), that's a much nicer experience than [get-unity/download/archive](https://unity3d.com/get-unity/download/archive).

**Pull request merged**